x-pack/plugins/observability/public/components/burn_rate_rule_editor/dependencies.tsx

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import {
+  EuiBadge,
+  EuiBasicTable,
+  EuiButtonEmpty,
+  EuiButtonIcon,
+  EuiComboBox,
+  EuiComboBoxOptionOption,
+  EuiForm,
+  EuiFormRow,
+  EuiPopover,
+  EuiSpacer,
+  EuiTableActionsColumnType,
+  EuiText,
+  EuiTitle,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import React, { useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { Rule } from '@kbn/triggers-actions-ui-plugin/public';
+import { isRight } from 'fp-ts/lib/Either';
+import {
+  SloRule,
+  useFetchSLOsWithBurnRateRules,
+} from '../../hooks/slo/use_fetch_slos_with_burn_rate_rules';
+import { Dependency, DependencyRT } from '../../../common/typings';
+import {
+  ALERT_ACTION,
+  HIGH_PRIORITY_ACTION,
+  LOW_PRIORITY_ACTION,
+  MEDIUM_PRIORITY_ACTION,
+} from '../../../common/constants';
+
+const ACTION_GROUP_OPTIONS = [
+  { value: ALERT_ACTION.id, label: ALERT_ACTION.name },
+  { value: HIGH_PRIORITY_ACTION.id, label: HIGH_PRIORITY_ACTION.name },
+  { value: MEDIUM_PRIORITY_ACTION.id, label: MEDIUM_PRIORITY_ACTION.name },
+  { value: LOW_PRIORITY_ACTION.id, label: LOW_PRIORITY_ACTION.name },
+];
+
+interface DependencyEditorProps {
+  isLoading: boolean;
+  onSubmit: (dependency: Dependency) => void;
+  dependency?: Dependency;
+  rules?: Array<Rule<SloRule>>;
+}
+
+function DependencyEditor({ isLoading, onSubmit, dependency, rules }: DependencyEditorProps) {
```

Comment:
Reminder to extract the component into its own file when we turn this POC into production code, but for now that's ok